apps/trading/client-pages/market/market-header-stats.tsx

```diff
@@ -79,27 +105,84 @@ export const MarketHeaderStats = ({ market }: MarketHeaderStatsProps) => {
         </HeaderStat>
       ) : null}
       <MarketLiquiditySupplied
-        marketId={market?.id}
+        marketId={market.id}
         assetDecimals={asset?.decimals || 0}
       />
-      <MarketProposalNotification marketId={market?.id} />
+      <MarketProposalNotification marketId={market.id} />
     </>
   );
 };
 
 type ExpiryLabelProps = {
-  market: Market | null;
+  market: Market;
+};
+
+export const FundingRate = ({ marketId }: { marketId: string }) => {
+  const { data: fundingRate } = useFundingRate(marketId);
+  return (
+    <div data-testid="funding-rate">
+      {fundingRate ? `${(Number(fundingRate) * 100).toFixed(4)}%` : '-'}
+    </div>
+  );
+};
+
+const padStart = (n: number) => n.toString().padStart(2, '0');
+
+export const FundingCountdown = ({ marketId }: { marketId: string }) => {
```

Comment:
This seems like a sensible thing to unit test. Perhaps split into two components so we can test the component logic without having to mock the data fetching